### PR TITLE
feat: added contact sales form

### PR DESCRIPTION
**Pull request by tan-90**

### CODE REVIEW

File: src/components/Billing/index.tsx [hunk 1]
```diff
@@ -43,68 +46,78 @@ export default function Billing() {
     recalculatePricing(sliderTable[sliderValue].value, paymentType)
   }, [])
 
+  useEffect(() => {
```
Comment (tan-90) [inline src/components/Billing/index.tsx]: Disable scrolling when the modal is open.

File: src/components/ContactSalesForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  async function handleSubmit(e) {
+    e.preventDefault()
+
+    setFormState(FormState.loading)
+
+    const { ok, error } = await fetch(submitEndpoint, {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
```
Comment (tan-90) [inline src/components/ContactSalesForm/index.tsx:30]: Payload format taken from https://github.com/fingerprintjs/fingerprintjs.com/blob/1ff179affb937fd75f431af64fe9ccd65943c17f/src/javascript/demo.ts#L133

File: src/components/Header/index.tsx [hunk 3]
```diff
@@ -8,47 +8,64 @@ import { ReactComponent as BurgerSvg } from './burger.svg'
...
 
 export default function Header() {
   const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false)
+  const [isContactSalesModalOpen, setIsContactSalesModalOpen] = useState(false)
 
   useEffect(() => {
     Prism.highlightAll()
   }, [])
 
+  useEffect(() => {
```
Comment (tan-90) [inline src/components/Header/index.tsx]: Disable scrolling when the modal is open.

File: src/components/MobileNavbar/index.tsx [hunk 4]
```diff
@@ -1,73 +1,87 @@
...
 import Button from '../common/Button'
 import classNames from 'classnames'
+import Modal from '../../components/common/Modal'
+import ContactSalesForm from '../../components/ContactSalesForm'
+import styles from './MobileNavbar.module.scss'
 
 export default function MobileNavbar() {
+  const [isContactSalesModalOpen, setIsContactSalesModalOpen] = useState(false)
+
+  useEffect(() => {
```
Comment (tan-90) [inline src/components/MobileNavbar/index.tsx]: Disable scrolling when the modal is open.

File: src/components/PriceCalculator/index.tsx [hunk 5]
```diff
@@ -16,6 +18,11 @@ export default function PriceCalculator() {
 
   const [selectedPreset, setSelectedPreset] = useState(selectOptions[0])
   const [customCount, setCustomCount] = useState<number | undefined>(undefined)
+  const [isContactSalesModalOpen, setIsContactSalesModalOpen] = useState(false)
+
+  useEffect(() => {
```
Comment (tan-90) [inline src/components/PriceCalculator/index.tsx]: Disable scrolling when the modal is open.

File: src/components/common/Modal/Modal.module.scss [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+  justify-content: center;
+  align-items: center;
+
+  width: 100vw;
+  height: 100vh;
+
+  top: 0;
+  left: 0;
+
+  background-color: c('off-white');
```
Comment (tan-90) [inline src/components/common/Modal/Modal.module.scss]: Fallback.

File: src/components/MobileNavbar/index.tsx (hunk 4 above)
Comment (Lisenish) [inline src/components/MobileNavbar/index.tsx]: Why we have to copy this effect in every place? 
Maybe move it into the modal?

File: src/components/MobileNavbar/index.tsx [hunk 7]
```diff
@@ -1,73 +1,87 @@
...
-                <a href='https://twitter.com/FingerprintJs' target='_blank' rel='noreferrer'>
-                  <TwitterSvg />
-                </a>
-              </li>
-            </ul>
+    <>
+      <div className={styles.nav}>
+        <Container size='large' className={styles.container}>
+          <div className={classNames(styles.links, styles.top)}>
+            <Button onClick={() => setIsContactSalesModalOpen(true)} variant='outline'>
```
Comment (Lisenish) [inline src/components/MobileNavbar/index.tsx:21]: Maybe let's create Contact Sales custom button with Modal logic inside? To reduce copy paste logic

File: src/components/ContactSalesForm/index.tsx [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+import { FormState } from '../../types/FormState'
+import { sendEvent } from '../../utils/gtm'
+import classNames from 'classnames'
+import styles from './ContactSalesForm.module.scss'
+
+interface ContactSalesFormProps {
+  className?: string | string[]
+}
+export default function ContactSalesForm({ className }: ContactSalesFormProps) {
+  const submitEndpoint = process.env.FPJS_LEAD_URL
```
Comment (Lisenish) [inline src/components/ContactSalesForm/index.tsx]: We need to move it to env.ts, you also need to add GATSBY prefix in order to make it work

File: src/components/MobileNavbar/index.tsx (hunk 7 above)
Comment (Lisenish) [reply src/components/MobileNavbar/index.tsx:21]: Also looks like there is extra whitespace before Button tag.
We need to set up eslint and prettier extensions for you

File: src/components/PriceCalculator/index.tsx [hunk 9]
```diff
@@ -33,63 +40,69 @@ export default function PriceCalculator() {
...
+        </Column>
+        <Column title='Enterprise License'>
+          <div className={styles.description}>Enterprise support license with SLA</div>
 
-        <Button variant='outline' small href='mailto:sales@fingerprintjs.com'>
-          Contact Sales
-        </Button>
-      </Column>
-    </div>
+          <Button variant='outline' small onClick={() => setIsContactSalesModalOpen(true)}>
```
Comment (Lisenish) [inline src/components/PriceCalculator/index.tsx:92]: Also extra whitespace

File: src/components/common/Modal/index.tsx [hunk 10]
```diff
@@ -0,0 +1,41 @@
+import React, { useEffect, useRef } from 'react'
+import { isBrowser } from '../../../utils/detector'
+import { ReactComponent as CloseSvg } from '../../../img/close.svg'
+import styles from './Modal.module.scss'
```
Comment (Lisenish) [inline src/components/common/Modal/index.tsx:4]: Please add some reasonable padding to modal content so it has a little more "space"

File: src/components/common/Modal/index.tsx [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+      return () => {
+        document.removeEventListener('mousedown', onWrapperClicked)
+      }
+    }
+  }, [onClose, wrapperRef])
+
+  return open ? (
+    <div ref={wrapperRef} className={styles.wrapper}>
+      <div className={styles.modal}>
+        <header className={styles.header}>
```
Comment (Lisenish) [inline src/components/common/Modal/index.tsx]: Please add some reasonable padding to the header

File: src/components/ContactSalesForm/index.tsx [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+              value={website}
+              onChange={(e) => setWebsite(e.target.value)}
+              disabled={formState === FormState.loading}
+            />
+          </label>
+
+          <Button
+            className={styles.button}
+            type='submit'
+            mobileIcon={<ChevronRightSvg className={styles.mobileIcon} />}
```
Comment (Lisenish) [inline src/components/ContactSalesForm/index.tsx]: We don't need it to become an Icon on a mobile version it looks like there is enough space for text even on iPhone 5

**Review by Lisenish — CHANGES_REQUESTED** [review]

File: src/components/MobileNavbar/index.tsx (hunk 7 above)
Comment (tan-90) [reply src/components/MobileNavbar/index.tsx:21]: Weird, they are setup, I'll look into that.

image

File: src/components/ContactSalesForm/index.tsx (hunk 12 above)
Comment (tan-90) [reply src/components/ContactSalesForm/index.tsx]: That was just the common button component, I'll look into writing a custom class.

File: src/components/MobileNavbar/index.tsx (hunk 7 above)
Comment (tan-90) [reply src/components/MobileNavbar/index.tsx:25]: Just double-checked, they are active and formatting on save. I'll keep an eye out for this and see if it happens again.

Comment (tan-90) [reply src/components/MobileNavbar/index.tsx:21]: They weren't formatting on save. Fixed.

File: src/components/ContactSalesForm/ContactSalesForm.module.scss [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+    justify-content: center;
+    align-items: center;
+
+    border-radius: 1rem;
+    padding: 0 2rem;
+    width: 100%;
+
+    text-align: center;
+
+    @media screen and (max-width: 640px) {
```
Comment (Lisenish) [inline src/components/ContactSalesForm/ContactSalesForm.module.scss:34]: We need to fix this and other media queries after https://github.com/fingerprintjs/fingerprintjs.com/pull/40/files is merged

File: src/components/ContactSalesForm/index.tsx [hunk 14]
```diff
@@ -0,0 +1,105 @@
...
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        email,
+        website,
+        name: email,
+      }),
+    }).then((response) => response.json())
+
+    if (!ok) {
```
Comment (Lisenish) [inline src/components/ContactSalesForm/index.tsx:37]: Minor: For readability, it's better when you're main if branch based on positive condition, so you don't invert in mind when you're reading code.

File: src/components/ContactSalesForm/index.tsx [hunk 15]
```diff
@@ -0,0 +1,105 @@
...
+  const [email, setEmail] = useState('')
+  const [website, setWebsite] = useState('')
+  const { formState, errorMessage, updateFormState, updateErrorMessage } = useContext(FormContext)
+
+  async function handleSubmit(e) {
+    e.preventDefault()
+
+    updateFormState(FormState.loading)
+
+    const { ok, error } = await fetch(submitEndpoint, {
```
Comment (Lisenish) [inline src/components/ContactSalesForm/index.tsx:27]: Let's extract this API call to helper/api file and use here a concrete method like `await api.createNewLead`

File: src/components/ContactSalesForm/index.tsx [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+  const submitEndpoint = GATSBY_FPJS_LEAD_URL
+
+  const [email, setEmail] = useState('')
+  const [website, setWebsite] = useState('')
+  const { formState, errorMessage, updateFormState, updateErrorMessage } = useContext(FormContext)
+
+  async function handleSubmit(e) {
+    e.preventDefault()
+
+    updateFormState(FormState.loading)
```
Comment (Lisenish) [inline src/components/ContactSalesForm/index.tsx:25]: @borislobanov92 wanted to recommend using a generic hook for forms here. He will write his proposal

File: src/components/Header/index.tsx [hunk 17]
```diff
@@ -24,35 +27,45 @@ export default function Header() {
     Prism.highlightAll()
   }, [])
 
-  const handleToggleMobileMenu = () => {
+  const handleOpenMobileMenu = () => {
```
Comment (Lisenish) [inline src/components/Header/index.tsx:30]: Why it's renamed? The logic inside seems like a toggle for me

File: src/components/common/Modal/index.tsx [hunk 18]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    document.addEventListener('mousedown', onWrapperClicked)
+    return () => {
+      document.removeEventListener('mousedown', onWrapperClicked)
+    }
+  }, [onClose, wrapperRef])
+
+  useEffect(() => {
+    document.body.style.overflow = open ? 'hidden' : 'unset'
```
Comment (Lisenish) [inline src/components/common/Modal/index.tsx:28]: - It's for scrolling prevention, right? Probably good example where a comment would be useful :) 
- Probably doesn't matter for us but here is a potential bug if original overflow style on the body wasn't `unset`

**Review by Lisenish — CHANGES_REQUESTED** [review]
@tan-90 please take a loot at comments

File: src/components/Header/index.tsx (hunk 17 above)
Comment (tan-90) [reply src/components/Header/index.tsx:30]: This happened on a master merge, though I'm not sure why it was marked as a change in that case.

Comment (tan-90) [reply src/components/Header/index.tsx:30]: Reverted.

File: src/components/common/Modal/index.tsx (hunk 18 above)
Comment (tan-90) [reply src/components/common/Modal/index.tsx:28]: Added a comment. I thought about that potential bug, but I couldn't think about any other way to handle this, open to suggestions.

File: src/components/ContactSalesForm/ContactSalesForm.module.scss (hunk 13 above)
Comment (tan-90) [reply src/components/ContactSalesForm/ContactSalesForm.module.scss:34]: Should I add a `// TODO`?

File: src/components/ContactSalesForm/index.tsx (hunk 15 above)
Comment (tan-90) [reply src/components/ContactSalesForm/index.tsx:27]: Created `utils/api.ts`

File: src/components/ContactSalesForm/ContactSalesForm.module.scss (hunk 13 above)
Comment (Lisenish) [reply src/components/ContactSalesForm/ContactSalesForm.module.scss:34]: You can fix now

File: src/components/common/Modal/index.tsx (hunk 18 above)
Comment (Lisenish) [reply src/components/common/Modal/index.tsx:28]: Discussed in DM it's fine for now, we don't plan to change body overflow

**Pull request closed without merging**